src/java.base/share/classes/sun/net/www/protocol/https/AbstractDelegateHttpsURLConnection.java

```diff
@@ -332,10 +332,12 @@ private boolean useExtendedCallerInfo(URL url) {
         if (prop.startsWith("domain:")) {
             String[] domains = prop.substring(7).split(",");
             for (String domain : domains) {
-                if (target.equals(domain)) {
+                if (target.equalsIgnoreCase(domain)) {
                     return true;
                 }
-                if (domain.startsWith("*.") && target.endsWith(domain.substring(1))) {
+                String afterWildCard = domain.substring(1);
```

Comment:
`domain` could be an empty string if the property value is "domain:a,,b". I know it's invalid but at least let's try our best to avoid a runtime exception. In fact, why is this variable necessary? It looks like `regionMatches` allows you to compare ...er... regions.

Reply:
Done.